jhu-idc/idc-isle-dc
# Updates the File Media Type to only accept a few types.

This updates the File Media Type to only accept these file types: `tfw, zip, tar, tgz, gz`

This resolves https://github.com/jhu-idc/iDC-general/issues/340

The issue was to remove all the duplicate options for File and Document, so that it was clear which Media Type to use. 

For now we know we need File to accept `tfw` format.  I took the liberty to making File a little more flexible knowing that there might be times we need to upload a few zip/tar formats.  There was interest expressed in making File more of a catch all Media Type, so this takes us a little in that direction.  If these extra formats are unwanted, let me know and I'll take them out. 

- This updates the File Media Type to only accept the above formats
- This adds tests for the new file types
- This updates the asset image to include new file to test with. 
- This also updates the tests to change areas where a pdf or tif file was used as a File media type. 
- This updates the Image Derivative Context to only operate on Image Media Types.  Without this change, any media marked as Original File on a Node with a Model of Image would have a Thumbnail and Service file created for it, regardless of whether or not the Media Type was an image.   

### To test
Start the system and create a repository item. Add Media to the item and choose File Media Type. Ensure that you see the  mime types `tfw, zip, tar, tgz, gz` listed as what acceptable media are.   Add a file of those types and ensure that it is added successfully and that a FITS file is created for it. 

Run the tests - they test all the formats listed here. 

You can also try out a few manual tests of adding zip/tar/tfw files to the system. 

Resolves: https://github.com/jhu-idc/iDC-general/issues/340
